grudge/models/__init__.py

```diff
@@ -54,7 +54,7 @@ def estimate_rk4_timestep(self, actx, dcoll, t=None, fields=None):
         from grudge.dt_utils import characteristic_lengthscales
         import grudge.op as op
 
-        wavespeeds = self.max_characteristic_velocity(t, fields, dcoll)
+        wavespeeds = self.max_characteristic_velocity(actx, t, fields)
```

Comment:
Either:

* Update the interface spec of `max_characteristic_velocity`, or
* Make passing `fields` mandatory, so that you can extract the array context from that, or
* get the array context from one or the other, make both optional, with one required.

Reply:
I've updated the interface a bit here: https://github.com/inducer/grudge/pull/121/commits/697cc7bc37871415e7b62a3e0a7d8a8066ff256c I figured since all these operators will eventually be rewritten in a more functional form, it's probably not worth investing too much into the operator interfaces. So I just opted for a quick-and-easy solution